Snippets/Snippets_6/UpgradeGuides/5to6/CustomErrorHandlingBehaviour.cs

```diff
@@ -0,0 +1,59 @@
+namespace Snippets6.UpgradeGuides._5to6
+{
+    using System;
+    using System.Threading.Tasks;
+    using NServiceBus;
+    using NServiceBus.Pipeline;
+
+    #region 5to6-customerrorhandlingbehaviour
+    class CustomErrorHandlingBehaviour : Behavior<ITransportReceiveContext>
+    {
+        public override async Task Invoke(ITransportReceiveContext context, Func<Task> next)
+        {
+            try
+            {
+                await next().ConfigureAwait(false);
+            }
+            catch (MessageDeserializationException)
+            {
+                Console.WriteLine("CustomFaultManager - MessageDeserializationException");
+            }
+            catch (Exception)
+            {
+                Console.WriteLine("CustomFaultManager - ProcessingAlwaysFailsForMessage");
+                // do not rethrow the message if you want to mark the message as processed.
+                // rethrow to move the message to the error queue.
+                throw;
+                // if you want to rollback the receive operation instead of mark as processed:
+                //context.AbortReceiveOperation();
+            }
+        }
+    }
+    #endregion
+
+    #region 5to6-newmessageprocessingpipelinestep
+    class NewMessageProcessingPipelineStep : RegisterStep
+    {
+        public NewMessageProcessingPipelineStep()
+            : base("CustomErrorHandlingBehaviour", typeof(CustomErrorHandlingBehaviour), "Adds custom error behavior to pipeline")
+        {
+            InsertAfter("MoveFaultsToErrorQueue");
```

Comment:
Scenarios that I can think of (please add more)

* To get notified when message is retried or sent to error - Handled by the notifications
* Control the number of FLR's per message type/ex - Can this be done with a custom SLR policy?
* Control the number and delay of SLR's per message type/ex - Can this be done with a custom SLR policy?
* Send a given message to a different "error queue" - Add a behavior to the logical receive pipeline that try/catch and bus.Sends to the give queue

More?